src/java/org/apache/cassandra/utils/SystemInfo.java

```diff
@@ -0,0 +1,214 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.cassandra.config.CassandraRelevantProperties;
+import oshi.PlatformEnum;
+
+/**
+ * An abstraction of System information, this class provides access to system information without specifying how
+ * it is retrieved.
+ */
+public class SystemInfo
+{
+
+    private static long INFINITY = -1l;
+    private static long EXPECTED_MIN_NOFILE = 10000l; // number of files that can be opened
+    private static long EXPECTED_NPROC = 32768l; // number of processes
+    private static long EXPECTED_AS = 0x7FFFFFFFl; // address space
+    /**
+     * The default number of processes that are reported if the actual value can not be retrieved.
+     */
+    public static long DEFAULT_MAX_PROCESSES = 1024;
+
+    private Logger logger = LoggerFactory.getLogger(SystemInfo.class);
+
+    /* The oshi.SystemInfo has  the following note:
+     * Platform-specific Hardware and Software objects are retrieved via memoized suppliers. To conserve memory at the
+     * cost of additional processing time, create a new version of SystemInfo() for subsequent calls. To conserve
+     * processing time at the cost of additional memory usage, re-use the same {@link SystemInfo} object for future
+     * queries.
+     *
+     * We are opting for minimal memory footprint. */
+    private oshi.SystemInfo si ;
+
+    public SystemInfo() {
+        si = new oshi.SystemInfo();
+    }
+
+    /**
+     *
+     * @return The name of the current platform. (e.g. Linux)
+     */
+    public String platform()
+    {
+        return oshi.SystemInfo.getCurrentPlatform().name();
+    }
+
+    /**
+     * Gets the maximum number of processes the user can create.
+     * Note: if not on a Linux system this always return the
+     * @{code DEFAULT_MAX_PROCESSES} value.
+     * @return The maximum number of processes.
+     * @see #DEFAULT_MAX_PROCESSES
+     */
+    long getMaxProcess() {
+        // this check only works on Linux systems.
+        if (oshi.SystemInfo.getCurrentPlatform() == PlatformEnum.LINUX) {
+            Path p =  FileSystems.getDefault().getPath("/proc", Long.toString(getPid()), "limits");
+            try
+            {
+                List<String> lines = Files.readAllLines(p);
+                for (String s : lines)
+                {
+                    if (s.startsWith("Max processes"))
+                    {
+                        String[] parts = s.split("\s+");
+                        String limit = parts[2];
+                        if ("unlimited".equals(limit))
```

Comment:
cant be this just a simple ternary operator? `return "unlimited".equals(limit) ? INFINITY : Long.parse(limit);`